src/aws-cpp-sdk-core/source/http/crt/CRTHttpClient.cpp

```diff
@@ -0,0 +1,639 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/core/http/crt/CRTHttpClient.h>
+#include <aws/core/http/standard/StandardHttpResponse.h>
+#include <aws/core/http/standard/StandardHttpRequest.h>
+#include <aws/core/utils/ratelimiter/RateLimiterInterface.h>
+
+#include <aws/crt/http/HttpConnectionManager.h>
+#include <aws/crt/http/HttpRequestResponse.h>
+
+static const char *const CRT_HTTP_CLIENT_TAG = "CRTHttpClient";
+
+// Adapts AWS SDK input streams and rate limiters to the CRT input stream reading model.
+class SDKAdaptingInputStream : public Crt::Io::StdIOStreamInputStream {
+public:
+    SDKAdaptingInputStream(const std::shared_ptr<Utils::RateLimits::RateLimiterInterface>& rateLimiter, std::shared_ptr<Aws::Crt::Io::IStream> stream,
+        const Http::HttpClient& client, const Http::HttpRequest& request, bool isStreaming,
+        Aws::Crt::Allocator *allocator = Crt::ApiAllocator()) noexcept : 
+        Crt::Io::StdIOStreamInputStream(std::move(stream), allocator), m_rateLimiter(rateLimiter), 
+        m_client(client), m_currentRequest(request), m_isStreaming(isStreaming), m_chunkEnd(false) 
+    {
+    }
+protected:
+
+    bool ReadImpl(Crt::ByteBuf &buffer) noexcept override
+    {
+        if (!m_client.ContinueRequest(m_currentRequest) || !m_client.IsRequestProcessingEnabled())
+        {
+            return false;
+        }
+
+        bool isAwsChunked = m_currentRequest.HasHeader(Aws::Http::CONTENT_ENCODING_HEADER) &&
+            m_currentRequest.GetHeaderValue(Aws::Http::CONTENT_ENCODING_HEADER) == Aws::Http::AWS_CHUNKED_VALUE;

```

Comment:
trivial: do this check in the constructor, instead of every read call

Reply:
nice! will do